manageiq-operator/pkg/controller/manageiq/manageiq_controller.go

```diff
@@ -147,62 +146,83 @@ func (r *ReconcileManageIQ) generateHttpdResources(cr *miqv1alpha1.ManageIQ) err
 	}
 
 	if privileged {
-		httpdServiceAccount := miqtool.HttpdServiceAccount(cr)
-		if err := r.createk8sResIfNotExist(cr, httpdServiceAccount, &corev1.ServiceAccount{}); err != nil {
+		httpdServiceAccount, mutateFunc := miqtool.HttpdServiceAccount(cr, r.scheme)
+		if result, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, httpdServiceAccount, mutateFunc); err != nil {
 			return err
+		} else if result != "unchanged" {
```

Comment:
I think we should be using [`controllerutil.OperationResultNone`](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.4.0/pkg/controller/controllerutil/controllerutil.go#L122) rather than the string here.